apps/api/src/campaign-application/campaign-application.service.ts

```diff
@@ -1,24 +1,64 @@
-import { BadRequestException, HttpException, HttpStatus, Injectable } from '@nestjs/common'
+import { BadRequestException, Injectable, Logger } from '@nestjs/common'
 import { CreateCampaignApplicationDto } from './dto/create-campaign-application.dto'
 import { UpdateCampaignApplicationDto } from './dto/update-campaign-application.dto'
 import { PrismaService } from '../prisma/prisma.service'
+import { OrganizerService } from '../organizer/organizer.service'
+import { Person } from '@prisma/client'
 
 @Injectable()
 export class CampaignApplicationService {
-  constructor(private prisma: PrismaService) {}
+  constructor(private prisma: PrismaService, private organizerService: OrganizerService) {}
   async getCampaignByIdWithPersonIds(id: string): Promise<UpdateCampaignApplicationDto> {
     throw new Error('Method not implemented.')
   }
 
-  create(createCampaignApplicationDto: CreateCampaignApplicationDto) {
-    if (
-      !createCampaignApplicationDto.acceptTermsAndConditions ||
-      !createCampaignApplicationDto.transparencyTermsAccepted ||
-      !createCampaignApplicationDto.personalInformationProcessingAccepted
-    ) {
-      throw new BadRequestException('All agreements must be checked')
+  async create(createCampaignApplicationDto: CreateCampaignApplicationDto, person: Person) {
+    try {
+      if (
+        createCampaignApplicationDto.acceptTermsAndConditions === false ||
+        createCampaignApplicationDto.transparencyTermsAccepted === false ||
+        createCampaignApplicationDto.personalInformationProcessingAccepted === false
+      ) {
+        throw new BadRequestException('All agreements must be checked')
+      }
+
+      let organizer = await this.prisma.organizer.findUnique({
+        where: { personId: person.id },
+      })
+
+      if (!organizer) {
+        organizer = await this.organizerService.create({
+          personId: person.id,
+        })
+      }
+
+      const sanitizedData = {
```

Comment:
Prisma sanitizes the input by default. Seems like it also trims unnecessary spaces from user input, thus `sanitizedData` is not needed.